src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,524 @@
+    pub outside_press               : frp::Source,
+    pub cancel_project_name_editing : frp::Source,
+    pub project_name                : frp::Source<String>,
+    pub select_breadcrumb           : frp::Source<usize>
+}
+
+impl FrpInputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            push_breadcrumb             <- source();
+            pop_breadcrumb              <- source();
+            outside_press               <- source();
+            cancel_project_name_editing <- source();
+            project_name                <- source();
+            select_breadcrumb           <- source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb,outside_press,cancel_project_name_editing,project_name,
+            select_breadcrumb}
+    }
+}
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct DebugFrpInputs {
+    pub push_breadcrumb   : frp::Source<Option<LocalCall>>,
+    pub pop_breadcrumb    : frp::Source,
+    pub select_breadcrumb : frp::Source<usize>
+}
+
+impl DebugFrpInputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend!{ network
+            push_breadcrumb   <- source();
+            pop_breadcrumb    <- source();
+            select_breadcrumb <- source();
+        }
+        Self{push_breadcrumb,pop_breadcrumb,select_breadcrumb}
+    }
+}
+
+
+// ==================
+// === FrpOutputs ===
+// ==================
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct FrpOutputs {
+    pub breadcrumb_push   : frp::Source<Option<LocalCall>>,
+    pub breadcrumb_pop    : frp::Source,
+    pub project_name      : frp::Any<String>,
+    pub breadcrumb_select : frp::Source<(usize,Vec<Option<LocalCall>>)>
+}
+
+impl FrpOutputs {
+    /// Constructor.
+    pub fn new(network:&frp::Network) -> Self {
+        frp::extend! {network
+            breadcrumb_push   <- source();
+            breadcrumb_pop    <- source();
+            project_name      <- any_mut();
+            breadcrumb_select <- source();
+        }
+        Self{breadcrumb_push,breadcrumb_pop,project_name,breadcrumb_select}
+    }
+}
+
+
+
+// ===========
+// === Frp ===
+// ===========
+
+#[derive(Debug,Clone,CloneRef)]
+#[allow(missing_docs)]
+pub struct Frp {
+    pub inputs  : FrpInputs,
+    pub outputs : FrpOutputs,
+    pub debug   : DebugFrpInputs,
+    pub network : frp::Network,
+}
+
+impl Deref for Frp {
+    type Target = FrpInputs;
+    fn deref(&self) -> &Self::Target {
+        &self.inputs
+    }
+}
+
+impl Default for Frp {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Frp {
+    /// Constructor.
+    pub fn new() -> Self {
+        let network = frp::Network::new();
+        let inputs  = FrpInputs::new(&network);
+        let outputs = FrpOutputs::new(&network);
+        let debug   = DebugFrpInputs::new(&network);
+        Self{network,inputs,outputs,debug}
+    }
+}
```

Comment:
As I've written in previous reviews, lets keep the order:
1. Struct def
2. Derefs
3. Methods
4. Other impls